### PR TITLE
Fixes #39433 - Nullify ACS SSL credentials before destroying organization

### PR DESCRIPTION
**Pull request by vsedmik**
#### What are the changes introduced in this pull request?
Adds a before_destroy callback in OrganizationExtensions that nullifies ssl_ca_cert_id, ssl_client_cert_id, and ssl_client_key_id on any Alternate Content Source records referencing the organization's Content Credentials before the Content Credentials themselves are destroyed.

#### Considerations taken when implementing this change?
AlternateContentSource has no organization_id and is not directly owned by the organization, so it is not destroyed as part of the org deletion cascade. The fix introduced in #11737 changed the ACS associations on ContentCredential from dependent: :nullify to dependent: :restrict_with_exception, which correctly blocks standalone CC deletion but also broke org deletion. The callback is placed before has_many :gpg_keys in the included block to ensure it fires before the Content Credentials are destroyed.

#### What are the testing steps for this pull request?
1. Create a Content Credential of cert type.
2. Create an ACS using that credential as CA cert, client cert, or client key.
3. Delete the organization that owns the Content Credential -> deletion should succeed.
4. Verify the ACS still exists but its SSL cert references are now blank.

## Summary by CodeRabbit

* **Chores**
  * Enhanced organization deletion to properly clean up SSL certificate and client key references associated with alternate content sources.

* **Tests**
  * Added test cases to verify SSL credential cleanup during organization destruction and handling of edge cases.

### CODE REVIEW

File: app/models/katello/concerns/organization_extensions.rb [hunk 1]
```diff
@@ -154,6 +155,14 @@ def associate_default_http_proxy
           end
         end
 
+        def nullify_acs_ssl_credentials
```
Comment (ianballou) [inline app/models/katello/concerns/organization_extensions.rb:158]: I would prefer to do this via `on delete: nullify` in the DB, but we can't do that and preserve the safeguard that stops users from delete content credentials that are being used in an ACS.

Callbacks can be fun because using `delete` instead of `destroy` works around  them.

**Review by ianballou — APPROVED** [review]
Thanks!
